lupa/_lupa.pyx

```diff
@@ -535,6 +547,89 @@ def unpacks_lua_table_method(meth):
     return wrapper
 
 
+cdef int check_lua_stack(lua_State* L, int extra) except -1:
+    """Wrapper around lua_checkstack
+    On failure, a LuaMemoryError is raised
+    """
+    if not lua.lua_checkstack(L, extra):
+        raise LuaMemoryError("could not ensure %d free extra slots"
+                             " in the Lua stack" % (extra, ))
+    return 0
+
+
+cdef int get_object_length_aux(lua_State* L) nogil:
+    cdef size_t length = lua.lua_objlen(L, lua.lua_upvalueindex(1))
+    lua.lua_pushlightuserdata(L, <void*>length)
+    return 1
+
+
+cdef Py_ssize_t get_object_length(LuaRuntime runtime, lua_State* L, int index) except -1:
+    """Obtains the length of the object at the given valid index
+    If Lua raises an error, a LuaError is raised
+    If the object length doesn't fit Py_ssize_t, an OverflowError is raised
+    """
+    cdef int result
+    cdef size_t length
+    check_lua_stack(L, 1)
+    lua.lua_pushvalue(L, index)                        # value
+    lua.lua_pushcclosure(L, get_object_length_aux, 1)  # closure
+    result = lua.lua_pcall(L, 0, 1, 0)
+    if result:
+                                                       # err
+        raise_lua_error(runtime, L, result)            #
+                                                       # length
+    length = <size_t>lua.lua_touserdata(L, -1)
```

Comment:
Yes, they refer to the Lua stack. I always put one when the stack is changed or when we know something more about the stack. Maybe the confusion is on `lua_pcall`. We don't know until we evaluate the `result` variable what was pushed onto the stack: an error message, or the return values of the function just called. That's why I put `# err` before `raise_lua_error`, because it is not it who pushed `err`.